server/api/crud/runtimes/nuclio/function.py

```diff
@@ -248,6 +248,24 @@ def _compile_function_config(
     _resolve_and_set_base_image(function, config, client_version, client_python_version)
     function_name = _set_function_name(function, config, project, tag)
 
+    serving_spec = function._get_serving_spec()
+    if serving_spec is not None:
+        confmap_name = "model-conf-" + function_name
```

Comment:
this is error prone as function name + your prefix might exceed the k8s name limitation of 63 characters. 
I suggest moving this entire "system-generation" resources under k8s_helper functionality so we can later have a single place (class) to see all resources created by mlrun for mlrun purposes
the function (create_or_update) would receive config-map data and name (that we will truncate upon need)
the function would return the generated config map name, enrich it with a label that includes its original name.